src/inc/til/bitmap.h

```diff
@@ -454,7 +454,20 @@ namespace til // Terminal Implementation Library. Also: "Today I Learned"
         til::rectangle _rc;
         dynamic_bitset<> _bits;
 
-        mutable std::optional<std::vector<til::rectangle>> _runs;
+        // Memory pooling to save alloc/free work to the OS.
+        // The optional below will destroy the internal vector when it is reset.
+        // But that will do a whole free to the OS when we know we're probably going to rebuild
+        // it very shortly for the new set of runs. We could make a pair bool/vector
+        // or continue to use the optional with a PMR vector inside.
+        // It's static because we only need one pool to manage memory
+        // for all vectors of rectangles no matter which bitmap instance is making them.
+        static std::pmr::unsynchronized_pool_resource& _getPool() noexcept

```

Comment:
this will conflict with #8787; should we converge earlier rather than later?

Reply:
We decided over Teams to converge in this PR, because there's a conflict here anyway.